crates/nu-cmd-plugin/src/util.rs

```diff
@@ -91,6 +94,23 @@ pub(crate) fn modify_plugin_file(
     operate(&mut contents)?;
 
     // Save the modified file on success
+    // First, ensure the parent directory exists
+    if let Some(parent_dir) = plugin_registry_file_path.parent()
+        && !parent_dir.exists()
+    {
+        fs::create_dir_all(parent_dir).map_err(|err| {
+            IoError::new_with_additional_context(
+                err.not_found_as(NotFound::Directory),
+                file_span,
+                parent_dir.to_path_buf(),
+                "Cannot create plugin config directory. \
+                Ensure the parent path is accessible and you have write permissions"
```

Comment:
I don't think you have to write this long context. If you don't have the permissions to create such a dir, you already get a permissions denied error, making this very obvious. Very long messages in our messages get somewhat messy. You don't have to remove that, just wanna say. You can take a look how it looks in your terminal.